controllers/vmware/virtualmachinegroup_reconciler_test.go

```diff
@@ -0,0 +1,1227 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vmware
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	vmoprv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha2"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	clusterv1 "sigs.k8s.io/cluster-api/api/core/v1beta2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	vmwarev1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/vmware/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/vmoperator"
+)
+
+func Test_shouldCreateVirtualMachineGroup(t *testing.T) {
+	tests := []struct {
+		name            string
+		mds             []clusterv1.MachineDeployment
+		vSphereMachines []vmwarev1.VSphereMachine
+		want            bool
+	}{
+		{
+			name: "Should create a VMG if all the expected VSphereMachines exists",
```

Comment:
Let's add a unit test ensuring that a deleting vSphereMachine blocks.
I would suggest to use this test case as a setup and add an additional deleting machine

Reply:
@fabriziopandini 
Trying to understand this scenario, does it mean a deleting vSphereMachine won't be added to the VMGroup member list?

If yes, this case should cover it `name: "Should create a VMG if all the expected VSphereMachines exists, deleting MD should be ignored"`

Reply:
This is what is in the current implementation, and it makes sense to me because it makes sure that we start a placement decision only when the list of vSphereMachines is stable.

Let's please add a unit test and do not change the code